compiler/rustc_query_system/src/dep_graph/graph.rs

```diff
@@ -531,13 +531,14 @@ impl<K: DepKind> DepGraph<K> {
             // value to an existing node.
             //
             // For sanity, we still check that the loaded stable hash and the new one match.
-            if let Some(dep_node_index) = data.dep_node_index_of_opt(&node) {
-                let _current_fingerprint =
-                    crate::query::incremental_verify_ich(cx, data, result, &node, hash_result);
+            if let Some(prev_index) = data.previous.node_to_index_opt(&node)
+                && let Some(dep_node_index) = { data.current.prev_index_to_index.lock()[prev_index] }
```

Comment:
```suggestion
                // Wrap the call to `lock()` in braces to unlock it before proceeding.
                && let Some(dep_node_index) = { data.current.prev_index_to_index.lock()[prev_index] }
```

Reply:
You can't use braces like that to shorted lifetimes. I changed it to use nested `if`s instead.